grid-proxy/pkg/client/grid_client.go

```diff
@@ -72,114 +76,108 @@ func (g *Clientimpl) Ping() error {
 	if err != nil {
 		return err
 	}
+
 	if req.StatusCode != http.StatusOK {
 		return fmt.Errorf("non ok return status code from the the grid proxy home page: %s", http.StatusText(req.StatusCode))
 	}
+
 	return nil
 }
 
 // Nodes returns nodes with the given filters and pagination parameters
 func (g *Clientimpl) Nodes(filter types.NodeFilter, limit types.Limit) (res []types.Node, totalCount int, err error) {
-	query := nodeParams(filter, limit)
-	req, err := http.Get(g.url("nodes%s", query))
+	response, err := g.makeRequest("nodes", filter, limit)
 	if err != nil {
-		return
-	}
-	if req.StatusCode != http.StatusOK {
-		err = parseError(req.Body)
-		return
+		return nil, 0, err
 	}
-	if err := json.NewDecoder(req.Body).Decode(&res); err != nil {
+
+	if err := json.NewDecoder(response.Body).Decode(&res); err != nil {
 		return res, 0, err
 	}
-	totalCount, err = requestCounters(req)
+
+	totalCount, err = requestCounters(response)
+
 	return
 }
 
 // Farms returns farms with the given filters and pagination parameters
 func (g *Clientimpl) Farms(filter types.FarmFilter, limit types.Limit) (res []types.Farm, totalCount int, err error) {
-	query := farmParams(filter, limit)
-	req, err := http.Get(g.url("farms%s", query))
+	response, err := g.makeRequest("farms", filter, limit)
 	if err != nil {
-		return
+		return nil, 0, err
 	}
-	if req.StatusCode != http.StatusOK {
-		err = parseError(req.Body)
-		return
-	}
-	data, err := io.ReadAll(req.Body)
+
+	data, err := io.ReadAll(response.Body)
 	if err != nil {
 		return
 	}
-	err = json.Unmarshal(data, &res)
-	if err != nil {
+
+	if err = json.Unmarshal(data, &res); err != nil {
 		return
 	}
-	totalCount, err = requestCounters(req)
+
+	totalCount, err = requestCounters(response)
+
 	return
 }
 
 // Twins returns twins with the given filters and pagination parameters
 func (g *Clientimpl) Twins(filter types.TwinFilter, limit types.Limit) (res []types.Twin, totalCount int, err error) {
-	query := twinParams(filter, limit)
-	req, err := http.Get(g.url("twins%s", query))
+	response, err := g.makeRequest("twins", filter, limit)
 	if err != nil {
-		return
-	}
-	if req.StatusCode != http.StatusOK {
-		err = parseError(req.Body)
-		return
+		return nil, 0, err
 	}
-	data, err := io.ReadAll(req.Body)
+
+	data, err := io.ReadAll(response.Body)
 	if err != nil {
 		return
 	}
+
 	err = json.Unmarshal(data, &res)
```

Comment:
`if err = json.Unmarshal(data, &res); err != nil` would be good as you used before
please apply for others